src/differentiation.jl

```diff
@@ -25,57 +25,77 @@ differentiate(p, (x, y)) # should return [6xy+1, 3x^2+1]
 function differentiate end
```

Comment:
We should add the `Val{deg}` call above and say it is recommended as it can infer types in the vector case

Reply:
Done